airflow/jobs.py

```diff
@@ -1428,14 +1428,14 @@ def _process_dags(self, dagbag, dags, tis_out):
         """
         for dag in dags:
             dag = dagbag.get_dag(dag.dag_id)
-            if dag.is_paused:
-                self.log.info("Not processing DAG %s since it's paused", dag.dag_id)
-                continue
-
             if not dag:
                 self.log.error("DAG ID %s was not found in the DagBag", dag.dag_id)
                 continue
 
+            if dag.is_paused:
+                self.log.info("Not processing DAG %s since it's paused", dag.dag_id)
+                continue
+
```

Comment:
The order of checking `if not dag:` and `if dag.is_paused:` was improper here.

If dag is `None`, `if dag.is_paused:` will throw exception, and check `if not dag:` will never be invoked.